Marlin/Marlin_main.cpp

```diff
@@ -5004,22 +5085,22 @@ inline void gcode_M42() {
 
     bool stow_probe_after_each = code_seen('E');
 
-    float X_probe_location = code_seen('X') ? code_value_axis_units(X_AXIS) : X_current + X_PROBE_OFFSET_FROM_EXTRUDER;
+    float X_probe_location = code_seen('X') ? code_value_axis_units(X_AXIS) : X_current;
     #if DISABLED(DELTA)
-      if (X_probe_location < LOGICAL_X_POSITION(MIN_PROBE_X) || X_probe_location > LOGICAL_X_POSITION(MAX_PROBE_X)) {
+      if (X_probe_location - X_PROBE_OFFSET_FROM_EXTRUDER < LOGICAL_X_POSITION(MIN_PROBE_X) || X_probe_location - X_PROBE_OFFSET_FROM_EXTRUDER > LOGICAL_X_POSITION(MAX_PROBE_X)) {
         out_of_range_error(PSTR("X"));
         return;
       }
     #endif
 
-    float Y_probe_location = code_seen('Y') ? code_value_axis_units(Y_AXIS) : Y_current + Y_PROBE_OFFSET_FROM_EXTRUDER;
+    float Y_probe_location = code_seen('Y') ? code_value_axis_units(Y_AXIS) : Y_current;
     #if DISABLED(DELTA)
-      if (Y_probe_location < LOGICAL_Y_POSITION(MIN_PROBE_Y) || Y_probe_location > LOGICAL_Y_POSITION(MAX_PROBE_Y)) {
+      if (Y_probe_location - Y_PROBE_OFFSET_FROM_EXTRUDER < LOGICAL_Y_POSITION(MIN_PROBE_Y) || Y_probe_location - Y_PROBE_OFFSET_FROM_EXTRUDER > LOGICAL_Y_POSITION(MAX_PROBE_Y)) {
         out_of_range_error(PSTR("Y"));
         return;
       }
     #else
-      float pos[XYZ] = { X_probe_location, Y_probe_location, 0 };
+      float pos[XYZ] = { X_probe_location - X_PROBE_OFFSET_FROM_EXTRUDER, Y_probe_location - Y_PROBE_OFFSET_FROM_EXTRUDER, 0 };
       if (!position_is_reachable(pos, true)) {
```

Comment:
`position_is_reachable` with `true` as the second parameter already subtracts these offsets.